### PR TITLE
refactor: revamp rpc `starkNet_extractPrivateKey `

### PR DESCRIPTION
**Pull request by khanti42**
This PR is to move starkNet_extractPrivateKey rpc from root folder to rpcs folder and adopt the sharable rpc handler from this PR https://github.com/Consensys/starknet-snap/pull/320

This PR also rename the RPC to starkNet_displayPrivateKey

### Requirement:
Requires this PR to be merged first:
- [x] #322

### CODE REVIEW

File: packages/starknet-snap/src/rpcs/displayPrivateKey.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+} from '../utils';
+import { getAddressKeyDeriver } from '../utils/keyPair';
+import { getKeysFromAddress } from '../utils/starknetUtils';
+
+export const DisplayPrivateKeyRequestStruct = assign(
+  object({
+    address: AddressStruct,
+  }),
+  AuthorizableStruct,
+  BaseRequestStruct,
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/rpcs/displayPrivateKey.ts:21]:
```suggestion
```

File: packages/starknet-snap/src/rpcs/displayPrivateKey.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+> {
+  protected requestStruct = DisplayPrivateKeyRequestStruct;
+
+  protected responseStruct = DisplayPrivateKeyResponseStruct;
+
+  /**
+   * Execute the display private key request handler.
+   *
+   * @param params - The parameters of the request.
+   * @returns null.
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/rpcs/displayPrivateKey.ts:49]:
```suggestion
```

File: packages/starknet-snap/src/rpcs/displayPrivateKey.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  DisplayPrivateKeyResponse
+> {
+  protected requestStruct = DisplayPrivateKeyRequestStruct;
+
+  protected responseStruct = DisplayPrivateKeyResponseStruct;
+
+  /**
+   * Execute the display private key request handler.
+   *
+   * @param params - The parameters of the request.
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/rpcs/displayPrivateKey.ts:48]:
```suggestion
   * @param params - The parameters of the request.
   * @param params.address - The account address.
```

File: packages/starknet-snap/src/rpcs/displayPrivateKey.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+export class DisplayPrivateKeyRpc extends AccountRpcController<
+  DisplayPrivateKeyParams,
+  DisplayPrivateKeyResponse
+> {
+  protected requestStruct = DisplayPrivateKeyRequestStruct;
+
+  protected responseStruct = DisplayPrivateKeyResponseStruct;
+
+  /**
+   * Execute the display private key request handler.
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/rpcs/displayPrivateKey.ts:46]:
```suggestion
   * Execute the display private key request handler.
   * The private key will be display via a confirmation dialog.
```

File: packages/starknet-snap/src/rpcs/displayPrivateKey.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    const components = [text('Do you want to export your private key?')];
+
+    if (!(await confirmDialog(components))) {
+      throw new UserRejectedRequestError() as unknown as Error;
+    }
+
+    const alertComponents = [
+      text('Starknet Account Private Key'),
+      copyable(userPrivateKey),
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/rpcs/displayPrivateKey.ts:77]:
```suggestion
      copyable(this.account.privateKey),
```

File: packages/starknet-snap/src/rpcs/displayPrivateKey.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    params: DisplayPrivateKeyParams,
+  ): Promise<DisplayPrivateKeyResponse> {
+    const { address } = params;
+    const keyDeriver = await getAddressKeyDeriver(snap);
+    const { privateKey: userPrivateKey } = await getKeysFromAddress(
+      keyDeriver,
+      this.network,
+      await getStateData(),
+      address,
+    );
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/rpcs/displayPrivateKey.ts:67]:
```suggestion
```

**Review by stanleyyconsensys — COMMENTED** [review]
please rewrite the whole unit test with sample in signMessage

**Review by khanti42 — COMMENTED** [review]
LGTM

**Review by stanleyyconsensys — APPROVED** [review]
LGTM

**Pull request merged**